libs/core/src/main/java/org/opensearch/OpenSearchException.java

```diff
@@ -288,7 +291,10 @@ public List<String> getHeader(String key) {
         return headers.get(key);
     }
 
-    protected Map<String, List<String>> getHeaders() {
+    /**
+     * Returns the map of header keys and values.
+     */
+    public Map<String, List<String>> getHeaders() {
```

Comment:
I'm always nervous about opening up such widely used classes like this. If you really do need a map, it can be built using the existing API today with `exception.getHeaderKeys().stream().collect(Collectors.toMap(k -> k, exception::getHeader))`. However, I _think_ this is probably okay because we already expose all this information via those existing APIs, and furthermore the mutable collection reference is modifiable today via that `getHeaderKeys()` method, so I don't think there is a lot of risk here. I'll let anyone else weigh in as well if they have comments.